decompose/group_ica.py

```diff
@@ -0,0 +1,420 @@
+                         max_iter=max_iter, conv_eps=conv_eps,
+                         complex_mixing=complex_mixing,
+                         hamming_data=hamming_data,
+                         remove_outliers=remove_outliers,
+                         cost_function=cost_function,
+                         verbose=verbose)
+
+
+    # ------------------------------------------
+    # save results to disk
+    # ------------------------------------------
+    groupICA_obj = {'fn_list': fn_list,
+                    'W_orig': W_orig,
+                    'A_orig': A_orig,
+                    'quality': quality,
+                    'icasso_obj': icasso_obj,
+                    'fourier_ica_obj': fourier_ica_obj}
+
+
+    if stim_id:
+        fn_base = "group_FourierICA_%02ddB.obj" % (stim_id)
+    else:
+        fn_base = "group_FourierICA_resting_state.obj"
+
+    fnout = join(dirname(dirname(fn_list[0])), fn_base)
+    filehandler = open(fnout, "wb")
+    pickle.dump(groupICA_obj, filehandler)
+    filehandler.close()
+
+    # return dictionary
+    return groupICA_obj, fnout
+
+
+
+
+# +++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#  get time courses of FourierICA components
+# +++++++++++++++++++++++++++++++++++++++++++++++++++++++
+def get_group_fourierICA_time_courses(groupICA_obj, stim_name=None,
+                                      stim_id=1, stim_delay=0):
+
+
+    """
+    Module to get time courses from the FourierICA components.
+    Note, to save memory time courses are not saved during group
+    ICA estimation. However, to estimate the time courses will
+    take a while.
+
+        Parameters
+        ----------
+        groupICA_obj: either filename of the group ICA object or an
+            already swiped groupICA object
+        stim_name: name of the stimulus channel. Note, for
+            applying FourierCIA data are chopped around stimulus
+            onset. If not set data are chopped in overlapping
+            windows
+            default: stim_names=None
+        stim_id: Id of the event of interest to be considered in
+            the stimulus channel. Only of interest if 'stim_name'
+            is set
+            default: event_id=1
+        stim_delay: stimulus delay in milliseconds
+            default: stim_delay=0
+    """
+
+
+    # ------------------------------------------
+    # import necessary modules
+    # ------------------------------------------
+    import numpy as np
+    from mne.externals.six import string_types
+    from scipy import fftpack
+
+
+    # ------------------------------------------
+    # test if 'groupICA_obj' is a string or
+    # already the swiped object
+    # ------------------------------------------
+    if isinstance(groupICA_obj, string_types):
+        from pickle import load
+        filehandler = open(groupICA_obj, "rb")
+        groupICA_obj = load(filehandler)
+        filehandler.close()
+        fn_list = groupICA_obj['fn_list']
```

Comment:
I guess this should be outside the if condition.. 
'''
+        fn_list = groupICA_obj['fn_list']
+        nfn_list = len(fn_list)
'''